src/valkey-benchmark.c

```diff
@@ -352,6 +361,108 @@ static void freeServerConfig(serverConfig *cfg) {
     zfree(cfg);
 }
 
+void resetPlaceholders(void) {
+    if (placeholders.indices[0])
+        zfree(placeholders.indices[0]); /* indices are a single contiguous allocation */
```

Comment:
Oh, confusing!

Let's use a separate struct field for the indices continuous allocation and let each of them point into this one, i.e. even indices[0] can point to the beginning of the continuous allocation.

```suggestion
    if (placeholders.all_indices)
        zfree(placeholders.all_indices); /* indices are a single contiguous allocation */
```